### DISCUSSION TITLE
What to ask about a11y?

### DISCUSSION THREAD

**Discussion by LeaVerou** [Devographics/surveys#115] [State of HTML Design]
In the [State of HTML workshop](https://docs.google.com/document/d/1HFREcQNViR3hxYmAtoT1QicFSLROoZrXP3zbsFE_HrE/edit?resourcekey=0-aJldkMW8-UtOG4klRktQOA#heading=h.qst1ipcaif0u), we identified Accessibility as one of the focus areas we want to survey.
"Accessibility & content semantics" was also a focus area @chrishtr [had identified much earlier](https://github.com/Devographics/surveys/discussions/111#discussioncomment-6500417).

However, when I was working on the [feature proposals](https://github.com/Devographics/surveys/discussions/categories/state-of-html-2023-features), I struggled to come up with features belonging to this focus area without simply a long list of `aria-` attributes. So far, all I’ve got is:
- [`tabindex`](https://github.com/Devographics/surveys/discussions/113) (simply because it was one of the features we [removed](https://github.com/Devographics/surveys/issues/11#issuecomment-1228401749) from last year’s State of CSS)
- [ARIA as a general technology](https://github.com/Devographics/surveys/discussions/114). This seems too broad, but a list of `aria-*` attributes would get tiring, fast. Are there meaningful groups we could break this into?

It looks like it may be best to capture some of this via Part 2 questions (see [this comment](https://github.com/Devographics/surveys/discussions/115#discussioncomment-6554635))

**Comment by mfreed7** [comment]
We could ask questions about existing HTML features that are broken from an a11y front. For example, it has been [brought up](https://github.com/w3c/csswg-drafts/issues/8930#issuecomment-1646948967) that the `title` attribute does not produce accessible tooltips in current browsers. Similarly, we're working on improving the a11y for the `<details>`/`<summary>` elements. It'd be good to collect a list of those issues so we can prioritize and fix them.

**Comment by nitya** [comment]
I would love to ask questions about _how_ developers implement and validate accessibility compliance. Specifically
- Is accessibility testing part of your developer workflow - if so, what tools do you use?
- Is accessibility compliance part of your organizational goals - if so, how is it measured?
- From an authoring standpoint, what accessibility rules do you find hardest to implement?
- From a testing standpoint, do you perform manual assessments (e.g., validate screen reader usage)

Would these be considered in scope for the survey?

**Reply by LeaVerou** [reply]
These are all great, I wonder how we could ask about them without adding four new freeform questions to the survey.

**Reply by mikequiapos** [reply]
There's two ways to possibly go about asking this. Let me try this one. Here's the ask:

How does your organization incorporate accessibility into its developer workflow, including testing practices, compliance goals, and challenges faced during implementation and testing?

--------

How does your organization incorporate accessibility testing into its developer workflow, prioritize accessibility compliance as part of its goals, measure and assess the level of accessibility achieved, address the challenges faced during implementation and testing, and conduct manual assessments like validating screen reader usage?

A) We integrate accessibility testing into our developer workflow using specific tools like [Tool A], prioritize accessibility compliance as a significant organizational goal, measure and assess accessibility levels through [Measurement Method], address challenges in implementing specific accessibility rules such as [Challenging Rule], and conduct manual assessments, including validating screen reader usage as part of our accessibility testing process.

B) We do not currently incorporate accessibility testing into our developer workflow or prioritize accessibility compliance as an organizational goal.

C) We incorporate accessibility testing into our developer workflow, but we do not utilize specific tools for this purpose. We also do not have a structured approach to measure and assess the level of accessibility achieved. We face challenges in implementing specific accessibility rules such as [Challenging Rule], but we do not conduct manual assessments like validating screen reader usage.

D) We prioritize accessibility compliance as an organizational goal, but we do not have a defined approach to incorporate accessibility testing into our developer workflow. We do not measure and assess the level of accessibility achieved or conduct manual assessments like validating screen reader usage.

E) Other (please specify): [Provide your response]

**Comment by boazsender** [comment]
Would love to ask a similar question to the 2020 MDN DNA:

"What are the biggest pain points for you when it comes to web accessibility?" from page 52 of the report: https://mdn.dev/archives/insights/reports/pdf/MDN-Web-DNA-Report-2020.pdf

<img width="902" alt="Screen shot of results from page 52 of the above linked PDF" src="https://github.com/Devographics/surveys/assets/122117/bc51f0f7-847d-4ca1-95cb-eb569af70c00">

Would also be interested in seeing a question about accessibility testing methodology, like "Do you perform accessibility testing in your projects?" and if so "What tools do you use for accessibility testing?" with options like:

* Using Voice Over on my computer
* Using JAWS on my computer
* Using NVDA on my computer
* Axe 
* Lighthouse
* Assistive labs
* Voiceover.js
* Guidepup

Might be missing some tools in this list, this was just from memory.

**Reply by LeaVerou** [reply]
Thank you so much Boaz! 
I already added the a11y testing question, thanks for listing options too!
The data on pain points is great too, was this an open-ended input or multiple choice?

**Comment by boazsender** [comment]
It was multiple choice. The screenshot above is of the responses to that question. Sorry I was unclear.
